onnx/defs/tensor/defs.cc

```diff
@@ -318,6 +318,17 @@ ONNX_OPERATOR_SET_SCHEMA(
             }
           } else if (shapeInput) {
             targetShapeProto.CopyFrom(*shapeInput);
+          } else if (hasInputShape(ctx, 1)) {
```

Comment:
Thanks for the improvement! I think we can simplify the whole inference by calling [getShapeInput](https://github.com/onnx/onnx/blob/3dafb54b11a496cd933de17fd303324d0686fc3b/onnx/defs/shape_inference.cc#L449) which combines these three modes of getting shape information, including the new one below.

Reply:
Thanks for the tip! That simplifies things quite a lot.